packages/x-data-grid/src/models/gridBaseSlots.ts

```diff
@@ -309,3 +309,7 @@ export type TooltipProps = CommonProps & {
   enterDelay?: number;
   title: React.ReactNode;
 };
+
+export type IconProps = CommonProps & {
+  fontSize?: 'small' | 'medium' | 'large' | 'inherit';
```

Comment:
I'm wondering if we should change this prop name to `size`? Would make more sense in the context of non-font icon sets.

```suggestion
  size?: 'small' | 'medium' | 'large' | 'inherit';
```

Reply:
It would make sense, but then we also need to create a replacement/wrapper for material-ui's `createSvgIcon()`.

Reply:
Fair enough, happy for you to decide if it's worth the change or not.

Reply:
I did the change in the last commit but I'm not sure which one is best.

Reply:
But that will force users to use the grid's custom `createSvgIcon` if they want to create an icon, and it will make it impossible to customize a grid icon by passing a material icon directly.

Reply:
> it will make it impossible to customize a grid icon by passing a material icon directly.

Hmm, I didn't think about this. Let's keep it compatible with Material UI icons then.
Eventually, we might need to type icons differently depending on the design system used.

Reply:
So I just noticed that the `<GridActionsCellItem icon={<EditIcon />} />` API is also an icon typed as `any` -_- This change is becoming too painful, I prefer to stick with a material-compatible API. I have reverted back to what I had initially.

That being said, I've added an `IconProps` type, but for some reason typescript is convinced that material icons have a `component: any` prop even though it doesn't complain about it anywhere else than when I try to use the icon in a `JSXConstructor<IconProps>` field, so the typecheck fails. So I'm starting to see why someone typed those things as `any` to start with.

![image](https://github.com/user-attachments/assets/49e782e4-8a49-46b6-95c6-2b134fcd3f32)
![image](https://github.com/user-attachments/assets/8458ae6e-c7ee-448f-b915-09166a3d9d83)

I'm also thinking we should remove `GridColumnUnsortedIcon` from the icons as it's not really an icon.

Reply:
I have re-typed material's `createSvgIcon` to something usable.

Reply:
I re-typed material's `createSvgIcon` to a type that's usable, but that leaves slots unable to accept material icons still. I'll type them as `any` in the docs, that's the best idea I can come up with at the moment.

![image](https://github.com/user-attachments/assets/eccf4c9f-7244-42e9-b2f7-1bbd1e34d654)

Reply:
> I'll type them as any in the docs, that's the best idea I can come up with at the moment.

Is this any better than `React.JSXElementConstructor<any>` we had in the first place?

Reply:
Internally yes because we can typecheck icons when we use them. For non-material users (in the future), also yes because they know what they need to implement to make the grid icons work. But for current material users, if they want to customize the slots with material icons, they'd need to pass `icon as any`.

Reply:
> I've added an IconProps type, but for some reason typescript is convinced that material icons have a component: any prop

`createSvgIcon` returns `typeof SvgIcon`:
https://github.com/mui/material-ui/blob/daf94785cb9a562cfc3b00a6fbf9a993f5780827/packages/mui-material/src/utils/createSvgIcon.d.ts#L3

`SvgIcon` is `OverridableComponent`:
https://github.com/mui/material-ui/blob/daf94785cb9a562cfc3b00a6fbf9a993f5780827/packages/mui-material/src/SvgIcon/SvgIcon.d.ts#L104

`OverridableComponent` adds the `component` prop:
https://github.com/mui/material-ui/blob/daf94785cb9a562cfc3b00a6fbf9a993f5780827/packages/mui-material/src/OverridableComponent/index.ts#L21

Reply:
And that prop is made optional only at this point: https://github.com/mui/material-ui/blob/daf94785cb9a562cfc3b00a6fbf9a993f5780827/packages/mui-material/src/SvgIcon/SvgIcon.d.ts#L110

As far as I can see, our 2 options are leave things as `any` in our codebase, or require material icons to be passed as `any`. Do you see any other option? Which do you think is the best? Imo, this PR is better than leaving `any` in our codebase.

Reply:
> And that prop is made optional only at this point: [mui/material-ui@daf9478/packages/mui-material/src/SvgIcon/SvgIcon.d.ts#L110](https://github.com/mui/material-ui/blob/daf94785cb9a562cfc3b00a6fbf9a993f5780827/packages/mui-material/src/SvgIcon/SvgIcon.d.ts#L110)

Right, but we don't use `SvgIconProps` anywhere.
`OverridableComponent` declares the `component` prop as required though: https://github.com/mui/material-ui/blob/daf94785cb9a562cfc3b00a6fbf9a993f5780827/packages/mui-material/src/OverridableComponent/index.ts#L21
And this is the reason we see the error in `iconSlots`.
It's weird though, because the `component` prop is not actually required on the icons components 🫠

Reply:
> or require material icons to be passed as any

We'll need to document this then, otherwise how would the users know it's not an issue on their side?

Reply:
I've added a note in the docs & the migration guide.